### PR TITLE
NMS-8550: Pollerd configuration files

### PR DESCRIPTION
**Pull request by indigo423**
Initialize structure for configuration reload events. Create initial documentation for Pollerd related configuration files and if a restart is required or not.
- JIRA: http://issues.opennms.org/browse/NMS-8550
- Bamboo: Do not worry, we add a link to our continuous integration system here

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+|===
+
+.Pollerd configuration file overview
+[options="header, autowidth"]
+|===
+| File                        | Restart Required | Reload Event | Description
+| `poller-configuration.xml`  | yes              | yes          | Restart is required in case new monitors are created or removed.
+                                                                  Reload Event loads changed configuration parameters of existing monitors.
+| `response-graph.properties` | no               | no           | Graph definition for response time graphs from monitors
+| `poll-outages.xml`          | ?                | ?            | ?
```
Comment (indigo423) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: Are these files related to Pollerd? What is the function? Is a restart required, is a change applied immediately or is a complete restart of opennms required?

File: opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+| `poller-config.properties`  | ?                | ?            | ?
```
Comment (indigo423) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: Are these files related to Pollerd? What is the function? Is a restart required, is a change applied immediately or is a complete restart of opennms required?

File: opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc (hunk 1 above)
Comment (agalue) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: - poller-configuration.xml

It is the main configuration file for Pollerd, and it can be reloaded using reloadDaemonConfig.

- poll-outages.xml

It is related with the scheduled outages feature. I'm not sure if it is reloaded automatically or not.

- response-graph.properties

It contains the graph templates for the metrics collected by Pollerd, but the file itself is not managed by Pollerd. Any graph related file is reloaded automatically, every time it is changed.

Comment (indigo423) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: Descriptions added 👍

File: opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+[options="header, autowidth"]
+|===
+| File                        | Restart Required | Reload Event | Description
+| `poller-configuration.xml`  | yes              | yes          | It is the main configuration file for _Pollerd_, and it can be reloaded using _reloadDaemonConfig_.
+                                                                  Restart is required in case new monitors are created or removed.
+                                                                  Reload Event loads changed configuration parameters of existing monitors.
+| `response-graph.properties` | no               | no           | It contains the graph templates for the metrics collected by _Pollerd_, but the file itself is not managed by _Pollerd_.
+                                                                  Any graph related file is reloaded automatically, every time it is changed.
+| `poll-outages.xml`          | ?                | ?            | It is related with the scheduled outages feature.
+| `poller-config.properties`  | ?                | ?            | ?
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: I doesn't look like this file is used anymore, but it's still part of the distribution.

**Comment by j-white** [comment]
Looks good, but I think we need to sort out the ?? before merging.

**Comment by mvrueden** [comment]
Hey Ulf, please check the OCA.

File: opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc (hunk 3 above)
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: `poll-outages.xml` can be reloaded using `uei.opennms.org/internal/schedOutagesChanged` as described in this thread http://opennms.530661.n2.nabble.com/Automated-Scheduled-outages-td6295452.html

Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/operation/daemon-config-files/pollerd.adoc]: I can confirm that `poller-config.properties` is no longer used, we'll remove it in NMS-8873.

**Comment by opennms-bamboo** [comment]
Welcome @web-flow.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @web-flow.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by mvrueden** [comment]
Hey Ulf please recheck the OCA.

**Comment by indigo423** [comment]
@j-white Added your review comments, updated branch to latest foundation-2016 and resolved merge conflicts. From my point of view ready to get merged.
